### PR TITLE
Implement Proper

### PR DESCRIPTION
**Pull request by DavidLuong98**
Implement Proper #284

I'm opened for some suggestions if it can be done differently or if there needs to be some adjustment to unit tests.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs [hunk 1]
```diff
@@ -313,6 +313,11 @@ public static FormulaValue Upper(IRContext irContext, StringValue[] args)
             return new StringValue(irContext, args[0].Value.ToUpper());
         }
 
+        public static FormulaValue Proper(IRContext irContext, StringValue[] args)
+        {
+            return new StringValue(irContext, CultureInfo.CurrentCulture.TextInfo.ToTitleCase(args[0].Value));
```
Comment (jcoc611-microsoft) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs:318]: @MikeStall should `IRContext` expose the parameterized CultureInfo?

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Proper.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+>> Proper("e. e. cummings")
```
Comment (lesaltzm) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Proper.txt:1]: strings with non ascii characters?

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs (hunk 1 above)
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs:318]: JS implementation was done with regex + using the current locale. Check in powerapps that this handles stuff like acronyms the same way. 
image

Comment (lesaltzm) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs:318]: It's available on EvalRunner. See Value() in LibraryText.cs as an example.

Comment (DavidLuong98) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs:318]: image
Not sure if i fully understand you. Did you mean something like this? Are we expecting "RSVP" to maintain as "RSPV" after the proper function? Let me know if i set this up wrong

Comment (DavidLuong98) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs]: I edited this and will resolve the comment. Let me know if you think this is not the correct way.

Comment (lesaltzm) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs]: I meant that the ToTitleCase function in C# would keep it as RSVP according to the docs https://docs.microsoft.com/en-us/dotnet/api/system.globalization.textinfo.totitlecase?view=net-6.0

So we have a difference in behavior here.

Comment (DavidLuong98) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs]: I made the change, let me know if you think this is ok

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Proper.txt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Proper.txt:1]: FYI - ran against JS tests and these all pass.

**Review by MikeStall — APPROVED** [review]
:shipit:

**Pull request merged**